tests/poll_guard_test.rs

```diff
@@ -0,0 +1,64 @@
+use dura::poll_guard::PollGuard;
+use dura::snapshots;
+use std::thread::sleep;
+use std::time::Duration;
+
+mod util;
+
+#[test]
+fn changed_file() {
+    let tmp = tempfile::tempdir().unwrap();
+    let mut repo = util::git_repo::GitRepo::new(tmp.path().to_path_buf());
+    repo.init();
+    repo.write_file("foo.txt");
+    repo.commit_all();
+
+    let mut pg = PollGuard::new();
+    assert!(!pg.dir_changed(repo.dir.as_path()));
+
+    sleep(Duration::from_secs_f64(1.5));
```

Comment:
Given how similar a lot of these tests start, a lot of this work could be moved into a macro

Reply:
alright, i did this one, but man, i had some unexpected problems with tmpdir, it seems that `drop()` would be called at the end of the macro. Oh, well, got it fixed.

Reply:
I believe that is because your macro is creating a block (ie when it expands everything is wrapped in braces) because you're using `{{` and `}} at the start/end. 

If you were to drop the second brace on each pair, it will inline each line when expanded, so everything will stay in scope until the end of the function, rather than the end of the macro. That'd also remove the need to `let repo` twice

Reply:
Nope. The second brace makes it a single statement. Notice how the last line of the macro "returns" a value?

Reply:
I'm aware that's what it was doing. My suggestion was to change that and flatten it into multiple inline statements. Looking back though, I think it is best as-is as otherwise the binding to `repo` gets hidden in the macro.